ios/brave-ios/Sources/Brave/Frontend/Browser/Search/SearchViewController.swift

```diff
@@ -441,415 +681,305 @@ public class SearchViewController: SiteTableViewController, LoaderListener {
+          Preferences.BraveSearch.braveSearchPromotionCompletionState.value
+            != BraveSearchPromotionState.maybeLaterUpcomingSession.rawValue
+            ? Strings.BraveSearchPromotion.braveSearchPromotionBannerMaybeLaterButtonTitle
+            : Strings.BraveSearchPromotion.braveSearchPromotionBannerDismissButtonTitle,
+          for: .normal
+        )
+        $0.secondaryButton.addTarget(
+          self,
+          action: #selector(onBraveSearchPromtionIgnoreButton),
+          for: .touchUpInside
         )
-        cell.backgroundColor = .clear
       }
 
       return cell
     case .findInPage:
-      let cell = tableView.dequeueReusableCell(withIdentifier: "default", for: indexPath)
-      cell.textLabel?.text = String(format: Strings.findInPageFormat, dataSource.searchQuery)
-      cell.textLabel?.textColor = browserColors.textPrimary
-      cell.textLabel?.numberOfLines = 2
-      cell.textLabel?.font = .systemFont(ofSize: 15.0)
-      cell.textLabel?.lineBreakMode = .byWordWrapping
-      cell.textLabel?.textAlignment = .left
-      cell.backgroundColor = .clear
+      let cell = collectionView.dequeueReusableCell(for: indexPath) as SearchFindInPageCell
+      cell.setTitle(dataSource.searchQuery)
+      return cell
+    case .openTabsAndHistoryAndBookmarks:
+      let cell = collectionView.dequeueReusableCell(for: indexPath) as SearchOnYourDeviceCell
+      let site = dataSource.siteData[indexPath.row]
+      cell.setSite(site, isPrivateBrowsing: dataSource.tabType.isPrivate)
 
       return cell
     }
   }
 
-  override public func tableView(
-    _ tableView: UITableView,
-    numberOfRowsInSection section: Int
-  ) -> Int {
-    guard let section = dataSource.availableSections[safe: section] else {
-      return 0
+  public func collectionView(
+    _ collectionView: UICollectionView,
+    viewForSupplementaryElementOfKind kind: String,
+    at indexPath: IndexPath
+  ) -> UICollectionReusableView {
+    let section = availableSections[indexPath.section]
+
+    if kind == UICollectionView.elementKindSectionHeader {
+      if let header =
+        collectionView
+        .dequeueReusableSupplementaryView(
+          ofKind: kind,
+          withReuseIdentifier: "search_header",
+          for: indexPath
+        ) as? SearchSectionHeaderView
+      {
+        switch section {
+        case .searchSuggestionsOptIn, .searchSuggestions, .braveSearchPromotion:
+          assertionFailure("no header for search suggestion")
+        case .findInPage:
+          header.setTitle(Strings.findOnPageSectionHeader)
+        case .openTabsAndHistoryAndBookmarks:
+          header.setTitle(Strings.searchHistorySectionHeader)
+        }
+        return header
+      }
+    } else if kind == UICollectionView.elementKindSectionFooter {
+      let footer =
+        collectionView
+        .dequeueReusableSupplementaryView(
+          ofKind: kind,
+          withReuseIdentifier: "in_your_device_footer",
```

Comment:
probably should setup constant's for these or introduce similar type-safe versions like we have for the cells